test/unit/ngsiv2/provisioning/device-registration_test.js

```diff
@@ -0,0 +1,344 @@
+                    './test/unit/ngsiv2/examples/contextAvailabilityRequests/registerIoTAgent1.json'))
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                iotAgentLib.clearAll(done);
+            });
+        });
+
+        it('should return a ENTITY_NOT_FOUND error', function(done) {
+            iotAgentLib.register(device1, function(error) {
+                iotAgentLib.getDevice('lightUnexistent', 'smartGondor', 'gardens', function(error, data) {
+                    should.exist(error);
+                    should.not.exist(data);
+                    error.code.should.equal(404);
+                    error.name.should.equal('DEVICE_NOT_FOUND');
+                    done();
+                });
+            });
+        });
+    });
+
+    describe('When a device is removed from the IoT Agent', function() {
+        beforeEach(function(done) {
+
+            nock.cleanAll();
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .post('/v2/entities?options=upsert')
+                .reply(204);
+
+            contextBrokerMock
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .post('/v2/entities?options=upsert')
+                .reply(204);
+
+            contextBrokerMock
+                .delete('/v2/registrations/6319a7f5254b05844116584d')
+                .reply(204, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                async.series([
+                    async.apply(iotAgentLib.clearAll),
+                    async.apply(iotAgentLib.register, device1),
+                    async.apply(iotAgentLib.register, device2)
+                ], done);
+            });
+        });
+
+        it('should update the devices information in Context Broker', function(done) {
+            iotAgentLib.unregister(device1.id, 'smartGondor', 'gardens', function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When the Context Broker returns an error while unregistering a device', function() {
+        beforeEach(function(done) {
+            nock.cleanAll();
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5